tikv-client/src/main/java/com/pingcap/tikv/util/RangeSplitter.java

```diff
@@ -122,6 +123,75 @@ private static int rightCompareTo(ByteString lhs, ByteString rhs) {
     return Comparables.wrap(lhs).compareTo(Comparables.wrap(rhs));
   }
 
+  /**
+   * Group by a list of handles by the handles' region id
+   *
+   * @param tableId Table id used for the handle
+   * @param handles Handle list
+   * @return <RegionId, HandleList> map
+   */
+  public TLongObjectHashMap<TLongArrayList> groupByHandlesByRegionId(long tableId, TLongArrayList handles) {
```

Comment:
This seems very similar to RangeSplitter. Consider move it there and slight refactor instead of copy most of the code.

Reply:
Complicated logic should never be copied and changed this way. The maintainer of the code never knows to fix both places once bug appear (since they are not even in one place!)
If code can be reuse, it's good. If not, at lease make them stay together in one class.

Reply:
My bad, was too lazy to refactor...